Objects/exceptions.c

```diff
@@ -2748,286 +2872,290 @@ unicode_error_adjust_end(Py_ssize_t end, Py_ssize_t objlen)
     return end;
 }
 
-#define _PyUnicodeError_CAST(PTR)   ((PyUnicodeErrorObject *)(PTR))
-#define PyUnicodeError_Check(PTR)   \
-    PyObject_TypeCheck((PTR), (PyTypeObject *)PyExc_UnicodeError)
-#define PyUnicodeError_CAST(PTR)    \
-    (assert(PyUnicodeError_Check(PTR)), _PyUnicodeError_CAST(PTR))
-
 
-static inline int
-check_unicode_error_type(PyObject *self, const char *expect_type)
+/*
+ * Get various common parameters of a Unicode Error object.
+ *
+ * The caller is responsible to ensure that 'self' is a PyUnicodeErrorObject,
+ * although this condition is verified by this function on DEBUG builds.
+ *
+ * Return 0 on success and -1 on failure.
+ *
+ * Parameters
+ *
+ *     obj          The retrieved underlying 'object'.
+ *     objlen       The 'object' length.
+ *     start        The clipped 'start' attribute.
+ *     end          The clipped 'end' attribute.
+ *     as_bytes     Indicate whether the underlying 'object' is a bytes object.
+ *
+ * The 'obj', 'objlen', 'start' and 'end' parameters may be NULL
+ * to indicate that the parameter does not need to be stored.
```

Comment:
Perhaps group these?
```suggestion
 * Output parameters:
 *
 *     obj          The 'object' attribute.
 *     objlen       The length of 'object'.
 *     start        The clipped 'start' attribute.
 *     end          The clipped 'end' attribute.
 *
 * Any of these may be NULL to indicate that the value does
 * not need to be stored.
 *
 * Input parameter:
 *     as_bytes     If 1, the error's 'object' attribute must be a bytes object
 *                    (i.e. the call is for a `UnicodeDecodeError`).
 *                  If 0, 'object' must be a string.
 *                  Raise TypeError if it has a wrong type.
 *
```

Reply:
That would probably be better for the reader!